src/edu/washington/escience/myria/parallel/Server.java

```diff
@@ -1125,63 +1135,72 @@ public long createView(
   }
 
   /**
-   * Create a function and register it in the catalog
+   * Create a function and register it in the catalog.
    *
    * @param name the name of the function
-   * @param definition the function definition (must be postgres specific)
+   * @param definition the function definition  - this is postgres specific for postgres and function text for python.
    * @param outputSchema the output schema of the function
+   * @param isMultivalued indicates if the function returns multiple tuples.
+   * @param lang this is the language of the function.
+   * @param binary this is an optional parameter for function for base64 encoded binary for function.
+   * @param workers list of workers on which the function is registered: default is all.
    * @return the status of the function
    */
   public String createFunction(
       final String name,
       final String definition,
       final String outputSchema,
+      final Boolean isMultivalued,
+      final FunctionLanguage lang,
+      final String binary,
       final Set<Integer> workers)
       throws DbException, InterruptedException {
-    String response = "Created Function";
+
+    String response = "";
     Set<Integer> actualWorkers = workers;
     if (workers == null) {
       actualWorkers = getWorkers().keySet();
     }
+    try {
 
-    /* Postgres specific syntax - Validate the command */
-    Pattern pattern = Pattern.compile("(CREATE FUNCTION)([\\s\\S]*)(LANGUAGE SQL;)");
-    Matcher matcher = pattern.matcher(definition);
+      Map<Integer, SubQueryPlan> workerPlans = new HashMap<>();
+      for (Integer workerId : actualWorkers) {
+        workerPlans.put(
+            workerId,
+            new SubQueryPlan(
+                new DbCreateFunction(
+                    EmptyRelation.of(Schema.EMPTY_SCHEMA),
+                    name,
+                    definition,
+                    outputSchema,
+                    isMultivalued,
+                    lang,
+                    binary,
+                    null)));
```

Comment:
If you get rid of `connectionInfo` for the  `DbCreateFunction` method (see other comment), you can get rid of that here.